### PR TITLE
Project employees

### PR DESCRIPTION
**Pull request by Shubham2301**
This PR comprises of the Project Dashboard.

A dashboard is created for each project where we can add Employees to that project and see the list of added employees.

### CODE REVIEW

File: resources/assets/js/components/ProjectDetailsComponent.vue [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+          <h3 class="col-sm-12"><strong>{{project.name}}</strong></h3>
+        </div>
+
+        <div class="info-section row">
+          <div class="current-cycle-info col-sm-6">
+            <label for=""><strong>Current Cycle:</strong></label>
+            {{"29/10/2018 to 02/10/2018"}}
+            <br>
+            <label for=""><strong>Estimated Effort:</strong></label>
+            {{"40"}}
```
Comment (rathorevaibhav) [inline resources/assets/js/components/ProjectDetailsComponent.vue]: Well if you're printing hard-coded values, don't think you need to embed inside curly braces

File: resources/assets/js/components/ProjectDetailsComponent.vue [hunk 2]
```diff
@@ -0,0 +1,188 @@
+<template>
+  <div class="header">
```
Comment (rathorevaibhav) [inline resources/assets/js/components/ProjectDetailsComponent.vue]: can we keep the indentation to 4 spaces?

File: resources/assets/js/components/ProjectDetailsComponent.vue [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+    <div class="card-body">
+      <div class="project-info container">
+        <div class="head-section row">
+          <h3 class="col-sm-12"><strong>{{project.name}}</strong></h3>
```
Comment (rathorevaibhav) [inline resources/assets/js/components/ProjectDetailsComponent.vue]:
```suggestion
          <h3 class="col-12"><strong>{{project.name}}</strong></h3>
```

File: resources/assets/js/components/ProjectDetailsComponent.vue [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        </div>
+
+        <div class="info-section row">
+          <div class="current-cycle-info col-sm-6">
+            <label for=""><strong>Current Cycle:</strong></label>
+            {{"29/10/2018 to 02/10/2018"}}
+            <br>
+            <label for=""><strong>Estimated Effort:</strong></label>
```
Comment (rathorevaibhav) [inline resources/assets/js/components/ProjectDetailsComponent.vue]: do we need to use `<label>` tag here?

File: resources/assets/js/components/ProjectDetailsComponent.vue [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+      <hr>
+      <div class="project-actions row">
+        <div class="col-sm-12 project-action">
+          <table class="table table-striped table-bordered">
+            <thead>
+              <tr>
+                <th>
+                  Employee
+                  <span @click="showAddEmployeeForm">
+                    <i class="btn btn-default fa fa-plus"></i>
```
Comment (rathorevaibhav) [inline resources/assets/js/components/ProjectDetailsComponent.vue]: Not sure how this is showing up but I'll prefer a cleaner structure,
```suggestion
<button type="button" class="btn btn-default">
    <i class="fa fa-plus"></i>
</button>
```

**Comment by Shubham2301** [comment]
@rathorevaibhav I have worked on your feedback. Please and review and let me know if anything else is required. @bohraji I have deployed the 2nd part of the feature in which the employees can see their own projects from their dashboards. 

Please consider that as well for the review.

File: database/migrations/2018_10_23_202033_create_employee_project_table.php [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            $table->primary(['employee_id', 'project_id']);
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
```
Comment (rathorevaibhav) [inline database/migrations/2018_10_23_202033_create_employee_project_table.php]: Just this:
```php
public function down()
{
    Schema::dropIfExists('questions');
}
```

File: resources/assets/js/components/ProjectDetailsComponent.vue [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+        <div class="card-body">
+            <div class="project-info container">
+                <div class="head-section row">
+                  <h3 class="col-12"><strong>{{project.name}}</strong></h3>
+                </div>
+
+                <div class="info-section mt-3 row">
+                    <div class="current-cycle-info col-sm-6">
+                        <strong>Current Cycle:</strong>
+                        29/10/2018 to 02/10/2018
```
[the remark this answers is not in the document]
Comment (Shubham2301) [reply resources/assets/js/components/ProjectDetailsComponent.vue]: We still need to work on this part of the module. So until then to give a complete look to the view, we are using hard coded dummy values

File: resources/assets/js/components/HR/Employee/EmployeeProjects.vue [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<template>
```
[the remark this answers is not in the document]
Comment (Shubham2301) [reply resources/assets/js/components/HR/Employee/EmployeeProjects.vue]: it was an empty component which was not being used anywhere, removed it for now.

**Comment by Shubham2301** [comment]
Hi @rathorevaibhav. Worked on the feedback. Please have a look.

**Comment by Shubham2301** [comment]
@rathorevaibhav I have resolved the conflicts.

File: resources/assets/js/components/ProjectDetailsComponent.vue (hunk 7 above)
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply resources/assets/js/components/ProjectDetailsComponent.vue]: Ok. Guess it won't be Production ready until then, right?

**Comment by rathorevaibhav** [comment]
@Shubham2301 I'm merging this. Have left a question for you in the review feedback.

**Pull request merged**